src/incubator/toast/index.tsx

```diff
@@ -0,0 +1,276 @@
+  }, []);
+
+  useDidUpdate(() => {
+    if (!visible) {
+      clearTimer();
+    }
+
+    toggleToast(visible);
+  }, [visible]);
+
+  const onDismiss = useCallback(() => {
+    clearTimer();
+    props.onDismiss?.();
+  }, [props.onDismiss]);
+
+  const positionStyle = useMemo(() => {
+    return {
+      position: 'absolute',
+      left: 0,
+      right: 0,
+      [position]: 0
+    } as Pick<ViewStyle, 'top' | 'left' | 'right' | 'bottom' | 'position'>;
+  }, [position]);
+
+  const onLayout = useCallback(event => {
+    const height = event.nativeEvent.layout.height;
+    if (height !== toastHeight) {
+      setToastHeight(height);
+    }
+  },
+  [toastHeight]);
+
+  const renderRightElement = () => {
+    // NOTE: order does matter
+    if (showLoader) {
+      const loaderColors = [Colors.rgba(Colors.white, 0.3), Colors.rgba(Colors.grey30, 0.7)];
+      return (
+        <ActivityIndicator size={'small'} testID={`${testID}-loader`} color={loaderColors[1]} style={styles.loader}/>
+        // <Loader size={Loader.sizes.SMALL} color={loaderColors} style={styles.loader} testID={`${testID}-loader`}/>
+      );
+    }
+
+    if (action) {
+      return (
+        <Button
+          link
+          style={styles.action}
+          color={Colors.grey20}
+          {...action}
+          labelStyle={Typography.bodySmallBold}
+          accessibilityRole={'button'}
+          activeBackgroundColor={Colors.grey70}
+          testID={`${testID}-action`}
+        />
+      );
+    }
+  };
+
+  const renderMessage = () => {
+    const textAlign = centerMessage ? 'center' : 'left';
+    return (
+      <View accessible={Constants.isIOS} style={styles.messageContainer}>
+        <Text
+          testID={`${testID}-message`}
+          // @ts-expect-error
+          ref={viewRef}
+          style={[styles.message, {textAlign}, messageStyle]}
+          accessibilityLabel={accessibilityMessage}
+        >
+          {message}
+        </Text>
+      </View>
+    );
+  };
+
+  const renderIcon = () => {
+    const tintColor = iconColor;
+    return <Image source={icon} resizeMode={'contain'} style={styles.icon} tintColor={tintColor}/>;
+  };
+
+  const renderToastContent = () => {
+    if (!_.isUndefined(children)) {
+      return children;
+    }
+
+    return (
+      <View style={[styles.toastContent, style, backgroundColor ? {backgroundColor} : undefined]}>
+        {renderIcon()}
+        {renderMessage()}
+        {renderRightElement()}
+      </View>
+    );
+  };
+
+  const renderAttachmentContent = () => {
+    if (renderAttachment) {
+      return <View pointerEvents={'box-none'}>{renderAttachment()}</View>;
+    }
+  };
+
+  const _renderAttachment = (positionStyle: object, zIndex?: number) => {
+    return (
+      <View style={[positionStyle, {zIndex}]} pointerEvents={'box-none'}>
+        {renderAttachmentContent()}
+      </View>
+    );
+  };
+
+  const renderToast = () => {
+    if (!visible && !isAnimating) {
```

Comment:
Why this check is here as well? Is there a scenario where it will not return on line 209 but will return here?

Reply:
You are right, It was copied like that from the private code and I see that we have the same issue there as well.